### PR TITLE
tests: Ensure IcingaApplication is initialized before adding config

### LINKED ISSUE #6105: Snapshot builds fail on livestatus tests
This issue only appears on the build server and seems only to hit Debian Jessie and Ununtu trusty.
I was not able to reproduce this locally on buster.
```
/usr/bin/ctest --force-new-ctest-process -j1
Test project /home/jenkins/workspace/icinga2-snapshot/deb-debian-jessie-1binary/arch/x86_64/icinga2/obj-x86_64-linux-gnu
[ . . . ]
      Start 84: livestatus-livestatus/hosts
84/89 Test #84: livestatus-livestatus/hosts .............................................***Failed    0.01 sec
Test setup error: std::exception: 'apply' cannot be used with type 'Service'

      Start 85: livestatus-livestatus/services
85/89 Test #85: livestatus-livestatus/services ..........................................***Failed    0.01 sec
Test setup error: std::exception: 'apply' cannot be used with type 'Service'

      Start 86: icinga_checkable-icinga_checkable_flapping/host_not_flapping
86/89 Test #86: icinga_checkable-icinga_checkable_flapping/host_not_flapping ............   Passed    0.01 sec
      Start 87: icinga_checkable-icinga_checkable_flapping/host_flapping
87/89 Test #87: icinga_checkable-icinga_checkable_flapping/host_flapping ................   Passed    0.01 sec
      Start 88: icinga_checkable-icinga_checkable_flapping/host_flapping_recover
88/89 Test #88: icinga_checkable-icinga_checkable_flapping/host_flapping_recover ........   Passed    0.01 sec
      Start 89: icinga_checkable-icinga_checkable_flapping/host_flapping_docs_example
89/89 Test #89: icinga_checkable-icinga_checkable_flapping/host_flapping_docs_example ...   Passed    0.01 sec

98% tests passed, 2 tests failed out of 89

Total Test time (real) =  17.58 sec

The following tests FAILED:
	 84 - livestatus-livestatus/hosts (Failed)
	 85 - livestatus-livestatus/services (Failed)
```
See also [here](https://build.icinga.com/job/icinga2-snapshot/job/deb-ubuntu-trusty-1binary/arch=x86_64/136/console) and [here](https://build.icinga.com/job/icinga2-snapshot/job/deb-debian-jessie-1binary/arch=x86/137/console)

### PR DESCRIPTION
**Pull request by lazyfrosch**
This is basically a workaround for fixture ordering in Boost 1.55.

It fixes builds for Debian jessie and Ubuntu trusty.

fixes #6105

### CODE REVIEW

**Comment by lazyfrosch** [comment]
Not sure of the code style in `IcingaApplicationFixture`, please review!

**Comment by lazyfrosch** [comment]
Package builds for this branch:
* https://build.icinga.com/job/icinga2-dev/job/deb-debian-jessie-1binary/5/
* https://build.icinga.com/job/icinga2-dev/job/deb-ubuntu-trusty-1binary/3/
* https://build.icinga.com/job/icinga2-dev/job/deb-debian-wheezy-1binary/10/

**Comment by Crunsher** [comment]
Currently won't work because livestatus-fixture is missing the include. Easy to fix but should we maybe rename `icingaapplication-fixture.cpp` to `icingaapplication-fixture.hpp` @dnsmichi @gunnarbeutner ?

**Comment by lazyfrosch** [comment]
Right, non-unity builds won't like that...

**Comment by lazyfrosch** [comment]
Updated my commits to include a proper header. Should be fine now.

File: test/icingaapplication-fixture.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#include "icingaapplication-fixture.hpp"
+
+using namespace icinga;
+
+static bool IcingaInitialized = false;
+
+IcingaApplicationFixture::IcingaApplicationFixture()
+{
+	if (! IcingaInitialized)
```
Comment (gunnarbeutner) [inline test/icingaapplication-fixture.cpp]: There should be no spaces after unary operators.

File: test/icingaapplication-fixture.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		InitIcingaApplication();
+}
+
+void IcingaApplicationFixture::InitIcingaApplication()
+{
+	BOOST_TEST_MESSAGE("Initializing Application...");
+	Application::InitializeBase();
+
+	BOOST_TEST_MESSAGE("Initializing IcingaApplication...");
+	IcingaApplication::Ptr appInst;
```
Comment (gunnarbeutner) [inline test/icingaapplication-fixture.cpp]: Just initialize the appInst variable here, i.e.:

`IcingaApplication::Ptr appInst = new IcingaApplication();`

**Comment by lazyfrosch** [comment]
Style updated

**Comment by gunnarbeutner** [comment]
Merged. :)

**Pull request merged**